### PR TITLE
Add a Makefile for easier local dev and add tests execution with coverage artefacts creation to pipelines

### PR DESCRIPTION
**Pull request by ttrenty**

### CODE REVIEW

**Comment by ttrenty** [comment]
The test coverage report generated by the pipeline can be found in its log (I can't find it in the Github GUI...) here it's https://github.com/PFA-Visualgo/Visualgo-PyPI/actions/runs/8002190016/artifacts/1265805283

**Comment by ttrenty** [comment]
I used the types from @Larwive  but he has his own PR for it #9

**Review by Larwive — DISMISSED** [review]

**Comment by ttrenty** [comment]
Alright we good I update the branch with the dev branch, you  can approve/review it again ^^"

**Review by Larwive — DISMISSED** [review]

File: Makefile
Comment (Fyzyy) [inline Makefile:1]: I'm not a fan of creating an environment for the user. He should be the one creating it, perhaps by explaining it in the documentation.

File: requirements.txt
Comment (Fyzyy) [inline requirements.txt:1]: Why visualgo is in requirement ??

Comment (ttrenty) [reply requirements.txt:1]: I agree that that's weird. It should be changed

File: Makefile
Comment (ttrenty) [reply Makefile:1]: Yes I think I will not use the venv in the Makefile and ask the user to enable one before using the Makefile

File: requirements.txt
Comment (ttrenty) [reply requirements.txt:1]: Removed!

File: Makefile
Comment (ttrenty) [reply Makefile:1]: Done!

**Review by Larwive — APPROVED** [review]

**Review by Fyzyy — APPROVED** [review]

**Pull request merged**